mpack-instance-manager/src/main/python/instance_manager/instance_manager.py

```diff
@@ -0,0 +1,500 @@
+            ranging from: mpack, mpack-instance, subgroup-name, module-name and map of component instance
+            AND with a filtering on each level
+
+  Granularity works only while names for all consecutive levels are specified.
+  Levels: mpack/instance/subgroup/module
+  E.g If only mpack and subgroup names are specified, the granularity will work only on mpack level,
+      though the subgroup filer will be applied. But if the instance name is specified also, than only granular output
+      of subgroup will be returned.
+
+  Components are provided as map with key as 'component type' and value as 'list of individual component instances
+  names' OR empty map for all component instances present
+  """
+  return parse_mpack_instances_with_filtering(mpack, mpack_instance, subgroup_name, module_name, components_map,
+                                              output_path=True)
+
+
+def parse_mpack_instances_with_filtering(mpack_name_filter, instance_name_filter, subgroup_name_filter,
+                                         module_name_filter, components_name_filter_map, output_conf_dir=False,
+                                         output_path=False):
+  instances = MpackInstance.parse_instances_with_filtering(os.path.join(ROOT_FOLDER_PATH, INSTANCES_FOLDER_NAME),
+                                                           mpack_name_filter,
+                                                           instance_name_filter, subgroup_name_filter,
+                                                           module_name_filter,
+                                                           components_name_filter_map)
+  full_json_output = build_json_output(instances, output_conf_dir=output_conf_dir, output_path=output_path)
+
+  granular_json_output = build_granular_output(full_json_output, mpack_name_filter, instance_name_filter,
+                                               subgroup_name_filter,
+                                               module_name_filter)
+
+  return granular_json_output
+
+
+def build_granular_output(json_output, mpack_name_filter, instance_name_filter, subgroup_name_filter,
+                          module_name_filter):
+  if mpack_name_filter:
+    json_output = json_output['mpacks'][mpack_name_filter]
+    if instance_name_filter:
+      json_output = json_output[MpackInstance.plural_name][instance_name_filter]
+      if subgroup_name_filter:
+        json_output = json_output['subgroups'][subgroup_name_filter]
+        if module_name_filter:
+          json_output = json_output[ModuleInstance.plural_name][module_name_filter]
+  return json_output
+
+
+def build_json_output_from_instances_dict(instances_dict, plural_name, output_conf_dir, output_path):
+  result = {}
+  for instance_name in instances_dict:
+    result[instance_name] = instances_dict[instance_name].build_json_output(output_conf_dir, output_path)
+
+  return {plural_name: result}
+
+
+# FIXME use mpack.json to determine category
```

Comment:
mpack.json already has category field. Can you fix this function and get rid of this FIXME?